addons/event_sale/models/event_registration.py

```diff
@@ -26,19 +26,38 @@ class EventRegistration(models.Model):
 
     @api.depends('sale_order_id.state', 'sale_order_id.currency_id', 'sale_order_line_id.price_total')
     def _compute_registration_status(self):
+        confirming_registrations = self.env["event.registration"]
         for so_line, registrations in self.grouped('sale_order_line_id').items():
             cancelled_so_registrations = registrations.filtered(lambda reg: reg.sale_order_id.state == 'cancel')
             cancelled_so_registrations.state = 'cancel'
             cancelled_registrations = cancelled_so_registrations | registrations.filtered(lambda reg: reg.state == 'cancel')
             if not so_line or float_is_zero(so_line.price_total, precision_digits=so_line.currency_id.rounding):
                 registrations.sale_status = 'free'
-                registrations.filtered(lambda reg: not reg.state or reg.state == 'draft').state = "open"
+                to_open_registrations = registrations.filtered(
+                    lambda reg: not reg.state or reg.state == "draft"
+                )
+                confirming_registrations |= to_open_registrations.filtered(
+                    lambda reg: reg.state == "draft"
+                )
+                to_open_registrations.state = "open"
             else:
                 sold_registrations = registrations.filtered(lambda reg: reg.sale_order_id.state == 'sale') - cancelled_registrations
                 sold_registrations.sale_status = 'sold'
                 (registrations - sold_registrations).sale_status = 'to_pay'
-                sold_registrations.filtered(lambda reg: not reg.state or reg.state in {'draft', 'cancel'}).state = "open"
+                to_open_registrations = sold_registrations.filtered(
+                    lambda reg: not reg.state or reg.state in {"draft", "cancel"}
+                )
+                confirming_registrations |= to_open_registrations.filtered(
+                    lambda reg: reg.state in {"draft", "cancel"}
+                )
+                to_open_registrations.state = "open"
                 (registrations - sold_registrations - cancelled_registrations).state = 'draft'
+        if confirming_registrations and not self.env.context.get("install_mode", False):
+            # running the scheduler for demo data can cause an issue where wkhtmltopdf runs
+            # during server start and hangs indefinitely, leading to serious crashes we
+            # currently avoid this by not running the scheduler, would be best to find the
+            # actual reason for this issue and fix it so we can remove this check
+            confirming_registrations._update_mail_schedulers()
```

Comment:
It should be ok for now, the only issue is that it's possible we'll do rendering for no reason. But there doesn't seem to be any forced commit or anything.

Only workaround we could find is to use commit hooks to only call this when it's effectively committed (so it won't trigger in front-end onchange calls). But that's crappy and not necessary for now.

Reply:
and what about doing it in <sale.order>.action_confirm ? (and maybe in other action of sale order if needed ?)

Reply:
Hello, thanks for the suggestion!
I do agree that having the fix in the compute method (even if it's stored), might introduce other issues.
I updated the fix with STD suggestion; I moved the fix to `action_confirm` method (did not find the necessity to add it somewhere else because AFAIK we always call the `action_confirm` method to update the sale order state to `sale`), I think indeed that it is safer.
What do you think?